### PR TITLE
README updated with proposed changes

### LINKED ISSUE #31017: Update README for better readability
**I want to update the README.md file for Gradle with some listed changes :**

- Add Gradle's participation in Hacktoberfest 2024 to Gradle's main repository.
- To add more structure for better readability
- Add relevant labels
- Update the logo of Gradle used in README as after checking the [Branding Guidelines](https://gradle.com/brand/) found that the logo used in dark mode is outdated.

**_Please assign this issue to me._**

### PR DESCRIPTION
**Pull request by abhirajadhikary06**
Issue: **Update README for better readability**
Fixes #31017 

### Context
This update to the README aims to enhance clarity and accessibility, providing users with structured information about Gradle’s features, installation, integration, and community support. Improved readability will benefit new users, helping them quickly find resources and understand Gradle’s capabilities. Additionally, seasoned users will appreciate the organization and centralization of relevant links and community engagement avenues.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md).
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of the [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g., the code was written by yourself or the original code is licensed under a license compatible with the Apache License 2.0.
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in the pull request so that additional changes can be pushed by the Gradle team.
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective.
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic.
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes.
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`.
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`.

### CODE REVIEW

File: README.md
Comment (oleg-nenashev) [inline README.md:10]: Please do not use single-line entries.
Also, ReadTheDocs logo is not applicable as we do not use that service

File: README.md
Comment (oleg-nenashev) [inline README.md:56]: Those resources may not be helpful here. I would rather put Gradle Cookbook or other locations in "Learn Gradle" from https://community.gradle.org/resources/

File: README.md
Comment (oleg-nenashev) [inline README.md:102]: This link is replaced by DPE university

File: README.md
Comment (oleg-nenashev) [inline README.md:86]: This is not an extensive lost and the choices are not, hmm, ... expected

File: README.md
Comment (oleg-nenashev) [inline README.md:92]: Should be a relative link if you want to use it

File: README.md
Comment (oleg-nenashev) [inline README.md:90]: It is a good CTA, but I am not sure whether you want to emphasize the core repo there. See https://community.gradle.org/contributing/ which needs to be referenced

**Review by oleg-nenashev — COMMENTED** [review]
Thanks @abhirajadhikary06 ! While I like where the change is going, for some sections it looks like a mishmash of the links with some clearly being not a fit to the context. Were you using a GAI service to generate it? 

At the same tome, I think it can be polished to a merge-able state

**Comment by abhirajadhikary06** [comment]
> Thanks @abhirajadhikary06 ! While I like where the change is going, for some sections it looks like a mishmash of the links with some clearly being not a fit to the context. Were you using a GAI service to generate it?
> 
> At the same tome, I think it can be polished to a merge-able state

>> Okay I will be rechecking the links and the contents. And yes I have used GAI . First I collected all links and informations from different sites of Gradle and then in basic English wrote the content of the file. Next I provided it to GAI for restructuring and refining.

File: README.md
Comment (abhirajadhikary06) [reply README.md:10]: Okay, then we are left with logos that are been currently used. 
I have turned the labels to multi-line entries.
```
[![Revved up by Develocity](https://img.shields.io/badge/Revved%20up%20by-Develocity-06A0CE?logo=Gradle&labelColor=02303A)](https://ge.gradle.org/scans) <br>
[![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/4898/badge)](https://bestpractices.coreinfrastructure.org/projects/4898) <br>
[![Join the Gradle Community Slack](https://img.shields.io/badge/Slack-Join%20the%20Community-brightgreen?style=flat&logo=slack)](https://gradle.org/slack-invite) <br>
[![Gradle Documentation](https://img.shields.io/badge/Docs-Gradle%20User%20Guide-blueviolet?style=flat&logo=read-the-docs)](https://docs.gradle.org/current/userguide/userguide.html) 
```

File: README.md
Comment (abhirajadhikary06) [reply README.md:102]: > This link is replaced by DPE university

[DPE University](https://dpeuniversity.gradle.com/app/catalog)** - Explore tutorials designed to get you started quickly.

File: README.md
Comment (abhirajadhikary06) [reply README.md:56]: Replacing them with Gradle Community Resource and Gradle Coobook.

- **[Gradle Community Resources](https://community.gradle.org/resources/)**: Discover a range of resources, tutorials, and guides to support your Gradle journey, from foundational concepts to advanced practices.
- **[Gradle Cookbook](https://cookbook.gradle.org/preface/)**: Access a practical collection of Gradle recipes and best practices to help you work efficiently with Gradle in various scenarios.

File: README.md
Comment (abhirajadhikary06) [reply README.md:86]: Can you please clarify what changes are required ?
I guess you are asking to add more repositories ?

File: README.md
Comment (abhirajadhikary06) [reply README.md:90]: The link has been added.
Gradle welcomes contributors of all experience levels! Whether you're fixing bugs, writing documentation, or implementing new features, there's a place for you in the [Gradle community](https://community.gradle.org/contributing/).

**Comment by oleg-nenashev** [comment]
When using GA, you should be definitely careful on what it suggests. Still, I think it can be refined to a mergeable state

**Comment by abhirajadhikary06** [comment]
> When using GA, you should be definitely careful on what it suggests. Still, I think it can be refined to a mergeable state

Thank you for your feedback! I completely agree that we should be cautious with GA's suggestions. I have refined the content can you please review it.

File: README.md
Comment (oleg-nenashev) [inline README.md:3]: Why did you invert it?

File: README.md
Comment (oleg-nenashev) [inline README.md:13]: I get this render in the dark mode 
Screenshot 2024-10-29 at 19 41 05

Line breaks between labels are not helpful, and the logo

File: README.md
Comment (oleg-nenashev) [inline README.md:50]: This section does not fit the top-level content, as written. Also the links are obsolete, see the new ones on gradle.org

```suggestion
```

File: README.md
Comment (oleg-nenashev) [inline README.md:62]: Not relevant in this context

File: README.md
Comment (oleg-nenashev) [inline README.md:75]: This is wrong guidelines as written. Not sure what is the intent.
There is `#community-support` for questions, but no tips sharing channel

File: README.md
Comment (oleg-nenashev) [inline README.md:83]: The link text was not updated

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]

File: README.md
Comment (abhirajadhikary06) [reply README.md:3]: Oh! Sorry for overlooking that.

File: README.md
Comment (abhirajadhikary06) [reply README.md:13]: Okay! I have removed those line breaks.

File: README.md
Comment (abhirajadhikary06) [reply README.md:50]: Okay I'm updating the content to match it's structure to the rest of the content... Also updating the links

Comment (abhirajadhikary06) [reply README.md:50]:
```
### 🛠 **Seamless IDE & CI Integration**

Gradle is built to work smoothly with a variety of Integrated Development Environments (IDEs) and Continuous Integration (CI) systems, providing extensive support for a streamlined workflow:

-   **Supported IDEs**: Quickly integrate Gradle with [Android Studio](https://docs.gradle.org/current/userguide/gradle_ides.html), [IntelliJ IDEA](https://docs.gradle.org/current/userguide/gradle_ides.html), [Eclipse](https://docs.gradle.org/current/userguide/gradle_ides.html), [NetBeans](https://docs.gradle.org/current/userguide/gradle_ides.html), and [Visual Studio Code](https://docs.gradle.org/current/userguide/gradle_ides.html).
-   **Continuous Integration**: Gradle easily connects with popular CI tools, including Jenkins, [GitHub Actions](https://docs.github.com/actions), [GitLab CI](https://docs.gitlab.com/ee/ci/), [CircleCI](https://circleci.com/), and more, to streamline build and deployment pipelines.
```
Check this if it's correct.
And the dedicated guide for Continuous Integration is not available currently.

File: README.md
Comment (abhirajadhikary06) [reply README.md:62]:
```suggestion
> 💬 Join our **#docs** channel on the [Gradle Community Slack](https://gradle.org/slack-invite) to share insights or ask questions as you learn.
```
Okay I have removed this

File: README.md
Comment (abhirajadhikary06) [reply README.md:83]: This (`CONTRIBUTING.md`) is the link Contributing guide **Gradle's main repo**. 
Do you want me to replace it with the `CONTRIBUTING.md` of **Gradle's community repo** ?

File: README.md
Comment (abhirajadhikary06) [reply README.md:75]: Modified to :
```suggestion
> **Quick Tip**: New contributors to Gradle projects are encouraged to ask questions in the Slack `#community-support` channel.
```

File: README.md
Comment (abhirajadhikary06) [reply README.md:83]: For now I am making a commit with the changes leaving this unresolved... As I get a comment back I'll be resolving it. 
And then I hope we are good to go.

Comment (abhirajadhikary06) [reply README.md]: @oleg-nenashev Made some  updates to Contribution Section. Hope now it aligns the structure you were asking for.

**Review by oleg-nenashev — APPROVED** [review]
Thanks a lot for the changes @abhirajadhikary06 ! I think it can be merged, it is definitely a net improvement compared to the previous version. I will wait for an extra review by @lkasso 

I marked this PR as `hacktoberfest-accepted`, so it should count toward the general Hacktoberfest goal. Let me not if it does not

File: README.md
Comment (lkasso) [inline README.md:37]: Is there a reason for removing some of these? Why Swift Apps but not Swift Libraries?

File: README.md
Comment (lkasso) [inline README.md:32]:
```suggestion
### 🚀 **Getting Started**
```

File: README.md
Comment (lkasso) [inline README.md:24]:
```suggestion
## 🐘 **Gradle Build Tool** 
```

File: README.md
Comment (lkasso) [inline README.md:100]:
```suggestion
<br><br>
```

File: README.md
Comment (lkasso) [inline README.md:98]:
```suggestion

```

File: README.md
Comment (lkasso) [inline README.md:73]:
```suggestion
- **Documentation**: The [Gradle User Manual](https://docs.gradle.org/current/userguide/userguide.html) covers everything from basic to advanced configurations.
```

File: README.md
Comment (lkasso) [inline README.md:84]:
```suggestion
- **Contribution Guide**: [Contribute](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md) to Gradle by submitting patches or pull requests for code or documentation improvements.
```

**Review by lkasso — CHANGES_REQUESTED** [review]

File: README.md
Comment (abhirajadhikary06) [reply README.md:37]: Now I have incorporated all of these.
```
- **Build Projects for Popular Languages and Frameworks**:
  - [Java Applications](https://docs.gradle.org/current/samples/sample_building_java_applications.html)
  - [Java Modules](https://docs.gradle.org/current/samples/sample_java_modules_multi_project.html)
  - [Android Apps](https://developer.android.com/studio/build/index.html)
  - [Groovy Applications](https://docs.gradle.org/current/samples/sample_building_groovy_applications.html)
  - [Kotlin Libraries](https://docs.gradle.org/current/samples/sample_building_kotlin_libraries.html)
  - [Scala Applications](https://docs.gradle.org/current/samples/sample_building_scala_applications.html)
  - [Spring Boot Web Apps](https://docs.gradle.org/current/samples/sample_building_spring_boot_web_applications.html)
  - [C++ Libraries](https://docs.gradle.org/current/samples/sample_building_cpp_libraries.html)
  - [Swift Apps](https://docs.gradle.org/current/samples/sample_building_swift_applications.html)
  - [Swift Libraries](https://docs.gradle.org/current/samples/sample_building_swift_libraries.html)
  ```

**Comment by abhirajadhikary06** [comment]
@lkasso I've made the changes.

**Review by lkasso — APPROVED** [review]

**Pull request merged**
